Helm: add commonLabels

ftrigari opened the pull request:
#### What this PR does
Every object created by the chart could have custom labels.


#### Which issue(s) this PR fixes or relates to

Fixes #9067

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/mimir?pullRequest=9742) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/mimir?pullRequest=9742) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/mimir?pullRequest=9742) it.</sub>

NickAnge commented:
Hello @ftrigari . Thanks a lot for your contribution !

The proposed change looks good overall and makes sense. However, I noticed that the `[lint-helm](https://github.com/grafana/mimir/actions/runs/11531912763/job/33349090304?pr=9742)` check is currently failing. It seems this is due to the change introducing an empty line when no labels are provided. 

Could you please address this issue to ensure the lint-helm check passes?

NickAnge commented on operations/helm/charts/mimir-distributed/templates/minio/create-bucket-job.yaml:
```diff
@@ -15,6 +15,7 @@ metadata:
     chart: {{ template "minio.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+    {{- include "mimir.commonLabels" (dict "ctx" . ) | nindent 4 -}}
```
Thanks for adding the if statement at _helpers.tpl. 🙌  This fixed the new empty line if commonLabels do not exist.

Now it fails for a similar reason here, because mimir.commonLabels are empty `{}` so its adding an empty line. Why are we adding this line here ?

ftrigari replied:
Hi @NickAnge ,
I appreciate your support to helping me for fixing the issues.
I forgot to edit this line of code when I fixed previous issues but with my last commit I have solved this problem.

I think is important to add commonLabels here as well because this job is created by Mimir chart and every object created by this chart must have commonLabels. 

At this moment, helm lint works fine, but i have got some issues with docs and tests. Have I missed something?

NickAnge replied:
Thanks @ftrigari . 
I do not see anything failing in the PR though 🤔 . Is it possible that it is happening at your fork only ? Have you rebased your fork ?

ftrigari commented:
Hi @NickAnge, I'm sorry but I messed up my fork  😓  I deleted all my old commits, updated the repo, committed my edited files, and then created a new PR.
The new PR is https://github.com/grafana/mimir/pull/10385
